commands/instances.go

```diff
@@ -702,3 +707,151 @@ func LoadSketch(ctx context.Context, req *rpc.LoadSketchReq) (*rpc.LoadSketchRes
 		AdditionalFiles:  additionalFiles,
 	}, nil
 }
+
+// ArchiveSketch FIXMEDOC
+func ArchiveSketch(ctx context.Context, req *rpc.ArchiveSketchReq) (*rpc.ArchiveSketchResp, error) {
+	// sketchName is the name of the sketch without extension, for example "MySketch"
+	var sketchName string
+
+	sketchPath := paths.New(req.SketchPath)
+	if sketchPath == nil {
+		sketchPath = paths.New(".")
+	}
+
+	sketchPath, err := sketchPath.Clean().Abs()
+	if err != nil {
+		return nil, fmt.Errorf("Error getting absolute sketch path %v", err)
+	}
+
+	// Get the sketch name and make sketchPath point to the ino file
+	if sketchPath.IsDir() {
+		sketchName = sketchPath.Base()
+		sketchPath = sketchPath.Join(sketchName + ".ino")
+	} else if sketchPath.Ext() == ".ino" {
+		sketchName = strings.TrimSuffix(sketchPath.Base(), ".ino")
+	}
+
+	// Checks if it's really a sketch
+	if sketchPath.NotExist() {
+		return nil, fmt.Errorf("specified path is not a sketch: %v", sketchPath.String())
+	}
+
+	archivePath := paths.New(req.ArchivePath)
+	if archivePath == nil {
+		archivePath = sketchPath.Parent().Parent()
+	}
+
+	archivePath, err = archivePath.Clean().Abs()
+	if err != nil {
+		return nil, fmt.Errorf("Error getting absolute archive path %v", err)
+	}
+
+	// Makes archivePath point to a zip file
+	if archivePath.IsDir() {
+		archivePath = archivePath.Join(sketchName + ".zip")
+	} else if archivePath.Ext() == "" {
+		archivePath = paths.New(archivePath.String() + ".zip")
+	}
+
+	if archivePath.Exist() {
+		return nil, fmt.Errorf("archive already exists")
+	}
+
+	archive, err := os.Create(archivePath.Clean().String())
```

Comment:
```suggestion
	archive, err := archivePath.Create()
```

Reply:
That's how it works right now, if it's a directory it saves inside that dir a `<sketch_name>.zip`, if it doesn't exists `.zip` is appended to the path, after all this it checks if there's a already an existing archive with that name.